org.eclipse.dawnsci.analysis.dataset/src/org/eclipse/dawnsci/analysis/dataset/SliceStreamSupport.java

```diff
@@ -0,0 +1,142 @@
+/*-
+ * Copyright 2017 Diamond Light Source Ltd.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Jacob Filik - initial API and implementation and/or initial documentation
+ *    Matthew Gerring - initial API and implementation and/or initial documentation
+ */
+package org.eclipse.dawnsci.analysis.dataset;
+
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import org.eclipse.dawnsci.analysis.dataset.slicer.SliceViewIterator;
+import org.eclipse.january.DatasetException;
+import org.eclipse.january.dataset.IDataset;
+import org.eclipse.january.dataset.ILazyDataset;
+
+/**
+ * This class provides slices as streams for Lazy Datasets for instance:
+ * 
+ * <pre>
+ * <code>
+final ILazyDataset lz = Random.lazyRand(64, 64, 100, 100);		
+List<Number> maxes = SliceStreamSupport.sliceStream(lz, 2, 3).map(set->set.max()).collect(Collectors.toList());
+  </code>
+ *</pre> 
+ * @see StreamSupport 
+ */
+public class SliceStreamSupport {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1118)